model/roles.go

```diff
@@ -238,13 +255,10 @@ func LoadRoleManifest(manifestFilePath string, releases []*Release) (*RoleManife
 				return nil, fmt.Errorf("Health check for role %s should have exactly one of url, command, or port; got %v", role.Name, checks)
 			}
 		}
-	}
 
-	if rolesManifest.Configuration == nil {
-		rolesManifest.Configuration = &Configuration{}
-	}
-	if rolesManifest.Configuration.Templates == nil {
-		rolesManifest.Configuration.Templates = map[string]string{}
+		if errs := validateRoleRun(role, &rolesManifest, declaredConfigs); len(errs) != 0 {
```

Comment:
Not so much checks as massaging the input. Starts at line 198 in this file (`// Normalize flight stage`).